pytorch_lightning/accelerators/accelerator.py

```diff
@@ -74,7 +74,8 @@ def setup(self, trainer: 'Trainer', model: LightningModule) -> None:
             model: the model to train
         """
         self.connect_training_type_plugin(self.training_type_plugin, model)
-        self.setup_optimizers(trainer)
+        if not self.training_type_plugin.setup_optimizers_after_dispatch:
```

Comment:
Hey @SeanNaren 
1. Won't it be better to move `self.setup_optimizers(trainer)` in dispatch directly or there will be a blocking part ?
2. If we stick with `setup_optimizers_after_dispatch`, it would be more clear to have `setup_optimizers_in_pre_dispatch` ?

Reply:
I never thought about it honestly, that's a really good point lol. Let me just move it and see what happens

Reply:
After #6506 I understand this piece of code a bit better.

Because Apex wraps optimizers, the optimizers have to be set before setting up the precision plugin.

So I'll update the doc strings making it clear that if you enable `setup_optimizers_in_pre_dispatch` that this may break APEX.

Reply:
total n00b question: is it possible to delay setting up both the optimizers and precision plugin to pre-dispatch to make apex work?